File: apps/gpu-text-rendering/src/features/document/limits.ts
/** Encoded PDF/GDOC input must fit a single Rust Vec in the 32-bit WASM address space.
 * Keep in sync with Rust limits::MAX_FILE_BYTES.
 * Decoded sections, individual images and GPU residency have separate budgets. */
export const maxDocumentFileBytes = 2 * 1024 * 1024 * 1024 - 1;

/** Shared message for local files, transferred buffers and streamed responses. */
export const documentFileLimitMessage = 'Document exceeds the 2 GiB file size limit';

File: apps/gpu-text-rendering/rust/document-format/src/limits.rs
/// Encoded input must fit one Vec in 32-bit WASM (its allocation cannot exceed isize::MAX).
/// Decoded geometry, image storage and GPU residency have independent budgets.
/// Keep in sync with src/features/document/limits.ts.
pub(crate) const MAX_FILE_BYTES: usize = 2 * 1024 * 1024 * 1024 - 1;
